### PR TITLE
Added README for Rack Instrumentation

### PR DESCRIPTION
**Pull request by KKelvinLo**
This PR adds documentation for OpenTelemetry-Ruby's Rack Instrumentation, which closes [#239](https://github.com/open-telemetry/opentelemetry-ruby/issues/239). 
The README describes the Rack gem, installation methods and its license, following the pattern set in instrumentation/sinatra for consistency.

### CODE REVIEW

File: instrumentation/rack/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# OpenTelemetry Rack Instrumentation
+
+The Rack instrumentation is a community-maintained instrumentation for the [Rack][rack-home] Ruby jobs system.
```
Comment (fbogsany) [inline instrumentation/rack/README.md:3]:
```suggestion
The Rack instrumentation is a community-maintained instrumentation for the [Rack][rack-home] Ruby web server interface.
```

File: instrumentation/rack/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+Or, if you use [bundler][bundler-home], include `opentelemetry-instrumentation-rack` in your `Gemfile`.
+
+## Usage
+
+To use the instrumentation, call `use` with the name of the instrumentation:
+
+```ruby
+OpenTelemetry::SDK.configure do |c|
+  c.use 'OpenTelemetry::Instrumentation::rack'
```
Comment (fbogsany) [inline instrumentation/rack/README.md:21]:
```suggestion
  c.use 'OpenTelemetry::Instrumentation::Rack'
```

File: instrumentation/rack/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The `opentelemetry-instrumentation-rack` gem source is [on github][repo-github], along with related gems including `opentelemetry-api` and `opentelemetry-sdk`.
+
+The OpenTelemetry Ruby gems are maintained by the OpenTelemetry-Ruby special interest group (SIG). You can get involved by joining us on our [gitter channel][ruby-gitter] or attending our weekly meeting. See the [meeting calendar][community-meetings] for dates and times. For more information on this and other language SIGs, see the OpenTelemetry [community page][ruby-sig].
+
+## License
+
+The `opentelemetry-instrumentation-rack` gem is distributed under the Apache 2.0 license. See [LICENSE][license-github] for more information.
+
+[rack-home]: https://github.com/mperham/rack
```
Comment (fbogsany) [inline instrumentation/rack/README.md:43]:
```suggestion
[rack-home]: https://github.com/rack/rack
```

**Review by fbogsany — CHANGES_REQUESTED** [review]

**Comment by fbogsany** [comment]
Thanks @KKelvinLo. Can you please add a reference to the examples here: https://github.com/open-telemetry/opentelemetry-ruby/tree/master/instrumentation/rack/example. See [this example for Sidekiq](https://github.com/open-telemetry/opentelemetry-ruby/pull/386/commits/ba9363f5b1542f03c3b5439ea411647fde196a9e).

**Comment by KKelvinLo** [comment]
Example added 👍

**Review by ericmustin — APPROVED** [review]
👍 lgtm

**Review by robertlaurin — APPROVED** [review]

**Review by fbogsany — APPROVED** [review]

**Comment by fbogsany** [comment]
Can you please rebase after #393 is merged @KKelvinLo ?

**Pull request merged**